src/Polar/derivatives.jl

```diff
@@ -181,28 +184,24 @@ function AutoDiff.adj_hessian_prod!(
     nmap = length(H.map)
 
     # Init seed
-    for i in 1:nmap
+    @inbounds for i in 1:nmap
```

Comment:
How fast it is on the GPU? Do you think it would be worth rewriting this for loop as a kernel?

Reply:
The problem is that this has to be done on the CPU. I think let's take it in and get a general idea of the performance. There are other places in the code that may need work. I have no idea what the performance bottlenecks will be on larger cases.

Reply:
Oh the hanging from yesterday was my fault because I used `CPU()` somewhere where it should've been `CUDADevice()`.